### PR TITLE
PR: Add sponsors section

### PR DESCRIPTION
**Pull request by dalthviz**
A preview:
imagen

### CODE REVIEW

File: content/contents.lr [hunk 1]
```diff
@@ -249,3 +249,57 @@ Download Spyder with Anaconda
...
 data_section: download
+---
+sponsors:
+
+#### sponsors ####
+title: Sponsors
+----
+description: Spyder is funded thanks to the generous support of
+----
+community_description: and the donations we have received from _**our**_ users around the world through [_Opencollective_](https://opencollective.com/spyder/):
```
Comment (ccordoba12) [inline content/contents.lr]: `_**our**_` -> `our`

`[_Opencollective_]` -> `[Open Collective]`

**Review by ccordoba12 — COMMENTED** [review]
Nice job @dalthviz, thanks!

I left a tiny comment then this will be ready.

File: content/contents.lr [hunk 2]
```diff
@@ -249,3 +249,57 @@ Download Spyder with Anaconda
...
+----
+opencollective: spyder
+----
+sponsors:
+
+##### sponsor #####
+name: Quansight
```
Comment (ccordoba12) [inline content/contents.lr]: Is it possible to remove the name and just leave the logo here?

File: content/contents.lr [hunk 3]
```diff
@@ -249,3 +249,57 @@ Download Spyder with Anaconda
...
+-----
+instagram: 
+-----
+github: 
+-----
+linkedin: 
+-----
+website: https://www.quansight.com/
+##### sponsor #####
+name: Numfocus
```
Comment (ccordoba12) [inline content/contents.lr]: And also here?

**Review by ccordoba12 — COMMENTED** [review]
I also have this small question.

File: content/contents.lr (hunk 2 above)
Comment (goanpeca) [reply content/contents.lr]: You mean in the actual rendered html page, right?

Comment (dalthviz) [reply content/contents.lr]: Yep, without the name, this is a preview:

imagen

Comment (ccordoba12) [reply content/contents.lr]: Much better, thanks! And could we also remove the link icon below those logos too?

Comment (goanpeca) [reply content/contents.lr]: Agreed with the boss!

**Review by ccordoba12 — APPROVED** [review]
Merging to see how it looks in Develop. Thanks @dalthviz!

**Review by CAM-Gerlach — COMMENTED** [review]
I like the overall idea, but there are a number of problems with the implementation, and thus I would recommend against merging to as is and instead either rolling back temporarily, or fixing the issues identified with a legacy branch of ``lektor-icon`` (e.g. ``v0.1`` or ``v1.0``) and have ``website-spyder`` track that branch (as it should anyway) until merging the overhaul and implementing it here.

Specifically, functional problems with changes: 

* The sponsors' logos don't do anything; i.e. they don't link to the websites of the sponsoring organizations like they should
* The logos continue to scale up with the page past the last viewport size breakpoint leading to jittering, and is inconsistent with how the other grids of cards (e.g. components) stay in a fixed max column width
* This should not included in the navbar; its too long and cluttered as it is
* Too much space between the "Spyder is funded..." line and the sponsor cards

With regard to the theme changes, which were committed directly to the ``master`` branch there without a PR or any review:

* If committed now, they will cause significant conflicts with the finished PR I'm about to push doing the complete refactoring, 
* Nearly all of this work on the ``lektor-icon`` is either unnecessary with the new changes or redundant/incompatible, and it should be possible to implement with minimal changes to the theme after the overhaul.
* Also, the vast majority of what it adds is redundant, duplicate and non-DRY HTML, CSS and model INI code copy-pasted from the ``team``/``member`` templates, flowblocks and style rules, respectively.

A far simpler design, made easy with my changes, would be just adding a wrapper flowblock, either to the theme or the site, to which itself can incorporate flowblocks, and then adding (in the ``contents.lr``) an existing team flowbock for the sponsor logo cards followed by an existing contents flowblock for the prose and OC button. With the overhaul, each flowblock now automatically hides the title section when those fields are left empty (thus they can display properly inside another flowblock), and the contents flowblock incorporates a button feature that can be easily used to embed the OC "Sponsor" SVG.

Except for the simple "wrapper" flowblock (which with my changes, could be added on the `spyder-website` side without modifying the theme, though no reason not to since it would likely be useful to others), all of this could be done by just changing the ``contents.lr`` here and nothing else.

**Comment by CAM-Gerlach** [comment]
Honestly, the best course of action might be to not spend further time on fixing these things now and instead just make ``spyder-website`` temporarily track a ``legacy`` branch of ``lektor-icon`` with the current changes there, while ``reset``-ing ``master`` back to where it was before, as I detailed in spyder-ide/lektor-icon#1 . Then, once that's merged, they could be addressed as part of updating the site to use the new version of the theme, or in a separate PR after that.

**Comment by ccordoba12** [comment]
Thanks for your work on this @dalthviz! Now this is in `production`.

**Pull request merged**